After review of DTB37628-corrected and DTB38033-corrected

kalaspuffar opened the pull request:
Hi @josteinaj 

This PR now has the changes required for the latest report of DTB38033.

* If there is no text at all the validation fails.
* Allow ```role```, ```aria-label```, and ```aria-labelled``` by in navigation documents
* Added navigation to roles (used in landmarks)
* Allow any ```epub:type``` on anchors (required for landmarks as they are navigation types)
* ```epub:type``` is no longer required for endnotes to changed to fetch them with role for ```nordic_opf_and_html_26b```
* New rule "[nordic291] Backmatter can only use roles (doc-afterword, doc-appendix, doc-colophon, doc-conclusion, doc-dedication, doc-epigraph, doc-epilogue)"
* Changed a report message for deprecation to assert instead 
* >Why was this highlighted in green for DTB38033? I.e. it got the class "success" instead of "error", probably a bug.
* ```lic``` is now allowed inside of anchor tags
* Don't require ```epub:type``` for level1 sections
* Don't require ```epub:type="z3998:author"``` on docauthor elements
* Removed special handling of title headings
* Bug fixes for footnotes and rearnotes
* Removed rule [nordic266] requiring footnotes to be wrapped with aside.
* New rule "[nordic291] Backmatter can only use roles (doc-afterword, doc-appendix, doc-colophon, doc-conclusion, doc-dedication, doc-epigraph, doc-epilogue, doc-glossary, doc-index)"
* Adding support for general asides
* Fixing issues with poem structure
* Allowing epub type for z3998:author in epub types
* Adding more prefixes that are default and need not be specified.
* Allowing backmatter sections without roles, and accepting a few more correct roles.


Best regards
Daniel

josteinaj commented on src/main/resources/xml/schema/2020-1/nordic2020-1.sch:
```diff
@@ -887,8 +887,18 @@
         <p>Rearnotes should not be used. Use endnotes instead.</p>
         <rule context="*">
             <let name="context" value="concat('(&lt;', name(), string-join(for $a in (@*) return concat(' ', $a/name(), '=&quot;', $a, '&quot;'), ''), '&gt;)')"/>
-            <report test="tokenize(@epub:type, '\s+') = ('rearnote', 'rearnotes')">[nordic290] Rearnotes are deprecated. Endnotes are required to be used instead. <value-of select="$context"/></report>
+            <assert test="tokenize(@epub:type, '\s+') = ('rearnote', 'rearnotes')">[nordic290] Rearnotes are deprecated. Endnotes are required to be used instead. <value-of select="$context"/></assert>
```
Won't this require either `rearnote` or `rearnotes` on all elements? The context is `*`, and the test checks for the presence of one of the two values. It was `report` so that this would be forbidden on all elements.

kalaspuffar replied:
Correct, realized it and will change it in the next PR :)

kalaspuffar commented:
Hi @josteinaj 

I have now fixed the issue you found and added fixes for all the issues I found while reviewing the corrected book DTB37628 that I received from @martinpub today.

Please review again.

Best regards
Daniel

josteinaj commented on src/main/resources/xml/schema/2020-1/nordic2020-1.sch:
```diff
@@ -518,7 +518,7 @@
             <let name="context" value="concat('(&lt;', name(), string-join(for $a in (@*) return concat(' ', $a/name(), '=&quot;', $a, '&quot;'), ''), '&gt;)')"/>
             <assert test="(parent::html:section)">[nordic204a] 'footnote' must have a parent section. <value-of select="$context"/></assert>
-            <assert test="not(following-sibling::html:*[not(local-name()='aside' and @epub-type='footnote')])">[nordic204a] 'footnote' must be placed at the end of a section.</assert>
+            <assert test="not(following-sibling::html:*[not(local-name()='aside' and @epub:type='footnote')])">[nordic204a] 'footnote' must be placed at the end of a section.</assert>
```
`@epub:type = 'footnote'` should be `tokenize(@epub:type, '\s+') = 'footnote'` so that it handles cases with multiple epub:type values.

kalaspuffar replied:
Great catch, Fixed that issue.

kalaspuffar commented:
Hi @josteinaj

I have now fixed the issue you found and added fixes for all the issues I found while reviewing the corrected book DTB38033.

Please review again.

Best regards
Daniel
